pages/project-version/middleware/index.js

```diff
@@ -110,8 +113,51 @@ const canComment = () => (req, res, next) => {
     .catch(next);
 };
 
+const getPreviousVersion = () => (req, res, next) => {
+
+  const versions = orderBy(req.project.versions.filter(pv => pv.id !== req.versionId),
+    v => moment(v.createdAt).format('YYYY-MM-DD'), ['desc']);
+
+  if (versions.length > 0) {
+    let prevVersion = versions.find(v => {
+      if (moment(req.version.createdAt).isAfter(v.createdAt)) {
+        return v;
+      }
+    });
+
+    req.api(`/establishments/${req.establishmentId}/projects/${req.projectId}/project-versions/${prevVersion.id}`)
+      .then(({ json: { data } }) => {
+        req.prevVersion = data;
+      })
+      .then(() => next())
+      .catch(next);
+  }
+
+};
+
+const getVersionChanges = () => (req, res, next) => {
+  if (req.prevVersion) {
+    const cvKeys = traverse(req.version.data, null, []);
```

Comment:
I would make the second and third arguments in `traverse` optional - then you won't need to provide them first time round

Reply:
done